### ISSUE TITLE
variable-name rule should support injecting constructors with angular

### ISSUE THREAD

**Issue by btesser** [palantir/tslint#457]
The following code makes the variable-name inspection fail, despite the fact it is a valid use case.  Any suggestions?

```typescript
class MyController {
  constructor(ExternalConstructor: IExternalConstructor) {
    var z: IExternal = new ExternalConstructor();
  }
}
```

**Comment by leeavital** [comment]
TSLint isn't aware of types, so it doesn't know that ExternalConstructor is being used a constructor. You could write a custom rule that rule that is aware of variables that are used as constructors and allows them to be uppercased.

**Comment by Delagen** [comment]
```
constructor(@Inject(Http) http:Http)
```
[typedef] expected parameter: 'Inject' to have a typedef

**Comment by leeavital** [comment]
@Delagen  this looks like an unrelated bug. Moving this to its own issue.

**Comment by leeavital** [comment]
https://github.com/palantir/tslint/issues/479

**Comment by btesser** [comment]
Is there any way to specify an argument as a constructor? Maybe by using
```
function (<Function()=>void>UserConstructor)
```
Or even
```
function (UserConstructor: typeof User)
```
Or some similar syntax.  The type definition itself should allow tslint to know it isn't a normal variable

**Comment by adidahiya** [comment]
We should be able to check in the constructor function block if a variable is used as a constructor, i.e. invoked with `new`.

Also I think the latest TS language services provide some access to type information, but I haven't looked into it too much yet.

**Comment by adidahiya** [comment]
I think this request is best now addressed by the pascal-case option for the `variable-name` rule. The above solutions seem overly complicated compared to that option, so I'm closing this out.
